rir/src/compiler/pir/instruction.h

```diff
@@ -1256,6 +1256,9 @@ class FLI(CastType, 1, Effects::None()) {
         }
         return type;
     }
+    bool promToValue() const {
+        return kind == Kind::Upcast && MkArg::Cast(arg(0).val());
```

Comment:
you could return `Mkarg*` (or null for false), that might be cleaner for the client...